stew/shims/tables.nim

```diff
@@ -9,4 +9,15 @@ template init*[A, B](T: type OrderedTableRef[A, B]): auto = newOrderedTable[A, B
 template init*[A](T: type CountTable[A]): auto = initCountTable[A]()
 template init*[A](T: type CountTableRef[A]): auto = newCountTable[A]()
 
+template mgetOrPutLazy*[A, B](t: Table[A, B], key: A, val: B): var B =
+  type R = B
+
+  proc setter(loc: var R): var R =
+    if loc == default(R):
+      loc = val
+    loc
+
+  setter(mgetOrPut(t, key, default(R)))
```

Comment:
this looks pretty odd - ie if `B` is an `array[10000, uint64]`, this will.. zero-init such an array, zeroinit another such array (`default(R)`) then compare all those values?